### ISSUE TITLE
Level discussion area

### ISSUE THREAD

**Issue by XeroOl** [thejoshwolfe/snakefall#36]
I think we should have somewhere to discuss levels. I really want to know if other people can solve my levels, but it seems wrong to chat about levels in the issues comment section. It would be great to have an issue/discord server/teamspeak/place to chat about levels and discuss level designs/inspiration.

**Comment by JoeltheFox** [comment]
Xero, I am playing through your levels now, and I think I found an unintended solution to Adventure.  Maybe an okay place to discuss is Discord? My handle is Joel#1401

**Comment by JoeltheFox** [comment]
By the way I discuss some levels in my [snake article](https://joelthefox.github.io/2019-08-21-Snake-Puzzle-Games/)

**Comment by JoeltheFox** [comment]
I'm glad Gooby started to add some new levels. I made a couple of slight alterations for fun:
[Ship Wreck Wrecked](https://wolfesoftware.com/snakefall/#level=HyRr4JK1&17&31?*Q04*H0111*N0*11*N0111*z0*z0*z0*z0*802*00*42*502*4022*G022*20*02*701*20*02*10*12*501*R2/b0?405&436&467&437&468&470&439&471&440&409&410&469&128&160&192&191&190&159/s2?0&31&62&93&124&155&186&217&248&279&310&341&372&403&434/s3?408&407&406/)
[The Birdcage (fixed) (unfixed)](https://wolfesoftware.com/snakefall/#level=HyRr4JK1&17&30?4*00101*901222*005*5010222*0122*00200012*10101*50100010011*501012201*40220020001*901200021011002*8011*001*002011122*001*1020112*0010002022202*8011101*001010500020001*201000101*401120001*401000101*10201222*201*101000101010002*D0100010100111221*401*20101000100222*E010101011*C01*10101010110*02*10110001*2010101011*20222*11011*101110*H1/s1?55/b0?414/s0?28/f0?94/f6?162/f2?231/f1?337/f8?433/f5?321/f4?88/f3?389/f7?445/)
[The Mill's power went out](https://wolfesoftware.com/snakefall/#level=HyRr4JK1&17&31?*F02*202*Q02*Q02*201*C200222022*C01*402*002*3011*301*Q01*Q0100220022002*0021002022*301*F01011*30100*01*A02*4041000*2101*F010100*0100111*D01*1011100101022*A01011100111000102*B01*0010111000112*B01*9012*9010*610*41/b3?112&143&174&205/b4?201&202&171/b2?368&337&306&275/b1?214&183&152&121&90&59&28/b0?76/s0?45&44&75&74/s2?493&494&495&464&433/s1?43/f3?518/f0?479/f7?113/f4?215/)

**Comment by jmdiamond3** [comment]
Thanks for these changes, Joel, they revealed to me some errors. 

I hope you don't mind but I replaced the original Abandon Ship with your new version, which I think is objectively better. 

I also made one fix to The Mill that was just a vestige of a previous version, but the end of it may pose a problem as you found an easier solution than the one I intended. I'm working on that now.

I looked at The Birdcage briefly and used yours to eliminate the leftmost column, but I haven't tried to solve your version yet.

**Comment by JoeltheFox** [comment]
Sounds good, I don't mind at all.

I was going to do a version of Mine Rescue but I figured that even finding the differences would be annoying. But I found a couple of things you might want to know: the level is possible without the pink 5-unit shape, and is possible without having to move the lower right orange pusher block.

**Comment by jmdiamond3** [comment]
Damn, I'm going through the Mine Rescue solution now and I do see the two 4-unit snakes can avoid using the block pusher. I like that a lot as a mechanism so I'm going to start working on a fix. I'm sure when I get to the pink lightning bolt shape I'll see what you mean and I'll fix that too.

I made changes to The Mill based on your alterations with the intent to narrow the viable solutions. I also changed the name just out of preference. Let me know if you notice any additional problems, and feel free to change the language next to my levels that reference you if you want.

**Comment by jmdiamond3** [comment]
I'm making incremental changes and updating the levels with each, so if you notice some fixed and others not, it's just that I'm doing them one at a time.

**Comment by JoeltheFox** [comment]
A smaller matter in Mine Rescue that I forgot to mention initially - I don't think the purple 1x1 and the orange 1x1 are needed in the bottom centre-left area

**Comment by jmdiamond3** [comment]
Thanks, I see both of those. I'll fix one but the other I'll just delete because it's fine without that detail.

I don't, however, see how you can complete Mine Rescue without the pink lightning bolt in the upper left. When a snake reaches that pink block, you'll have snakes of 4, 6, 6, and 9 units. If you send a snake back down into the mine, you could have 4, 6, 7, and 9 unit snakes. You can get all but the 9 unit snake in the area where the pink block is, and 11 units are needed to leave the area without help from the pink block, so it seems to me it will always be necessary.

Let me know if I'm missing something.

**Comment by jmdiamond3** [comment]
I believe the last two issues you brought up are fixed in the main puzzle, but the solution hasn't been updated. I'll wait for a response on the pink block issue before I take the time to create the solution again.

**Comment by JoeltheFox** [comment]
If the yellow snake holds up the door to that area, the blue 9 snake can eat the 2 fruit and get to 11, then go through without the help of the pink block

**Comment by JoeltheFox** [comment]
Oops, I might have mixed up colors. By yellow snake I meant 6-snake above. By the way, to ensure use of the orange pusher block, you'll need to prevent the 6-snake from coming down to push blocks in from the right (in your solution it's not necessary for the 6-snake to trap itself in the top right, it can go down below the stack of 3 1x1 blocks instead). I think you could shorten the bottom right area so there's no space to push left from

**Comment by jmdiamond3** [comment]
I see that solution without the pink block now and I'm pretty sure I know how to fix it.

For the pusher block issue, have you looked at the new version or are you using the solution? Because the solution isn't updated while the new version attempts to address this problem. I'm going to fix the pink block issue and then I'll revisit the pusher block to make sure I didn't disable one alternative solution only to create another.

**Comment by JoeltheFox** [comment]
Yeah in the latest version, a 6-snake can still push blocks into the well from the right instead of using the orange pusher. It looks like the update prevented 4-snakes from being able to do it

**Comment by jmdiamond3** [comment]
But isn't the 6-unit snake (S6) trapped outside of the mine at that point? I thought when he released the trio of single blocks at the top he can't re-enter.

I've made so many changes in so many browser tabs I'm not sure the latest version even works but I need to take a break. There's a chance it addresses and fixes everything but I won't be experimenting with it for awhile. Thanks for all the help, it's much appreciated.

**Comment by jmdiamond3** [comment]
I underestimated myself and ended up continuing work on the level. The current level is slightly different than the solution level but the solution shows all the major moves.

I really think I've fixed everything but you have me worried I'm missing something regarding that S6 pushing blocks down to the bottom. I still think he's stuck outside the mine until the blue snake is freed.

**Comment by jmdiamond3** [comment]
Last two things for today: I can't solve your Birdcage variation, and I can't solve Level 7 in Vertebrae. With both I just want to know for sure they are solvable, not the solution. I'd like to get them myself but I don't want to play forever not realizing they're impossible.

**Comment by JoeltheFox** [comment]
Yep, my birdcage variation and level 7 in Vertebrae are both solvable!
For the 6-unit snake, I'll post a "solution" link here for that part when I get home to show what I mean

**Comment by jmdiamond3** [comment]
OK your Birdcage was brilliant and I'm going to rethink mine because I have a feeling now there are too many possible solutions.

As for Vertebrae #7, I know it's not solvable and this is just the government running stress tests on me.

**Comment by jmdiamond3** [comment]
There's no need to send me that "solution" to demonstrate the S6 doing all the work on those blocks. I just saw what you meant but it doesn't matter because, by complete coincidence, I made that impossible an hour ago just because I thought the changes to the level looked better from an aesthetic standpoint.

Still, it would be great if you could confirm you can't see any significant alternative solutions to any one part of the level.

On another note, did the dozens of exits scattered throughout the "mine" signify anything to you? I meant them to look like resources for which the mine was built and then when the last fruit is eaten, they're like explosions and the snakes have to escape before the mine collapses. Just curious whether you picked up on either of those.

**Comment by JoeltheFox** [comment]
I took over from your solution around move 500 - [link](https://wolfesoftware.com/snakefall/#level=HyRr4JK1&33&55?*o04*N02*41011*00*21*101*M0211*241011000*11010001110*21*E02111444*01011000100010112014102*012*D02*014441*00110001010101110141002112*D02*510*11000101*3014100022*D021141*30144410001011101100141*H02*110*314441*60*0141*G021100110*51411001100010001441*F0211*2011*2014410011*301441*E02111*20110*01014410111*00*01441*D0211*701441011410111*001*141*C02110001*202014410011101*20*01441*B02111*602014410001101*501441*A021141*60201441*20*11001001441*90211441*6020144*21*00111*001441*90114441*3010020144*21*001100*31*901*041*602014110001100011*4011*602101*041010*420111*101002100102010011*5021001*041*801*002001002101*3011*40211011*04*A10*71*002201011*302*00*014441*00*41*701*5011*20510*010011441021001*00*01*9012022111*2011*40*01*4010*0101*30100111011141*10*0101200101*3012001011*701001410144410002*1101*1010120221100010100110010001011011101444100211100051000111*3010101000111*70110014441*00110*0100*01011*201000111*60110100014441011*50*010011*001110*0100102*10211010001444*310111*4011*20101441001020020221101000144411*0410*810010100101441001022020221101010144411*041*B01*0010144*012220*0211010101444*H12*210*D10*51/b1?1275/b2?1617&1616&1560&1505&1450&1449&1448/b6?1357/b7?1302&1247/b8?1361&1306&1251/b9?1192&1193/b10?1527/b11?1245&1190/b12?1579&1524&1469&1414/b13?1359&1304/b14?1521/b16?764&763&762/b0?1279&1224/b15?1201&1202&1203&1147&1196/b21?627&572&517&518&463/b18?267/b19?212/b20?157/b4?1339/b3?1176/s0?114&115&116/s3?113&112&111/s1?1473/s2?1478&1533&1532&1587&1588&1643&1642&1697&1752/f1?91/f10?1682/f2?1518/f5?1519/f9?1026/f0?1177/f3?831/f6?886/b5?322&323&46&47&48&49/f7?809/b22?810&424&369&370&371/f4?868/#replay=nmGTi8PB&3rru*3rddlddlllddrr0*2luurrrd*4ruurururruullurrddll3rrdruruldlulllddllur0ululllddlur3uruu0*0ur3ldlururul0rrdrdlldddld*2rur*0u*2rulld*0lddrrrddrdldd*Aluuu*1rd*1rur*0u*2rurrrddlddl3lldd*9rururrddrru0lllu*0ruuruullulullldddrrru*0rurr*1dlluuu3uluurruuruururrddrruru*1r0luuurrr*0drrururrrddrrru3uurrdll0urrddrruuulldlldlluulluururdrruurrddrrd3l0lldlldldlluuuluu*1rddld*0ruululll3ldrr1*4l3uruuluuru*0lulluurrruurrddrululldllldd*8ruu1llddluuruuu*0ru*4ruulu3u1u3rurururrru0dd*3ruulu1rruruulu3u0uururur1ruu0uu1lullurr3lllulullur0uurruulluluuulluul*0urrddrrddllddrrruullddrruullddrruull1r3ruulddr0dddr*1dldldlddrru*2rddlluuruu*2ldlddrru*2rddlluuruu*2ldlddrru*2rddll)

The exits did make me think of ore, but I didn't catch the explosion analogy. I think it adds a nice flavor.

**Comment by jmdiamond3** [comment]
I’m on my phone and as far as I know I can’t watch on a phone, but I know what it’s going to show. One of these times it’s going to actually be fixed and complete.

**Comment by jmdiamond3** [comment]
Now that I know you have a pretty good handle on this game, I should ask - is Temple solvable? I created Temple 2 because I loved Temple but didn't think there were enough fruit to get the snake through the spike spirals.

**Comment by JoeltheFox** [comment]
In The Mill, it is still solvable if you erase the portals (not sure if that was on your to-do list already). I also discovered that it is possible if you move the exit and bottom wall one space to the right!

Temple... may or may not be solvable ;)

**Comment by JoeltheFox** [comment]
By the way with this amount of discussion, would Discord be a better place? I know XeroOl and Patashu also use discord. For more broad discussions than snakefall, the thinky-puzzle-games server is good as well.

**Comment by jmdiamond3** [comment]
Just solved The Sawmill after moving the wall and exit at the bottom. Good catch, I think that makes the puzzle much more interesting.

I wouldn't mind if having the portal meant you needed to use the portal, which is how I think it is now. But I have wanted to experiment with making it impossible to get the blue snake out without using the portal. For now I'll just move the exit.

As for Discord, I've used it for a total of 5 minutes in my life so I may need some directions, but I'll give it a try later and send you a message when I do.

**Comment by JoeltheFox** [comment]
Hm. I guess you can't entirely avoid the portal, but you can still for example [block it](https://wolfesoftware.com/snakefall/#level=HyRr4JK1&29&31?*F02*202*Q02*Q02*201*C200222022*C01*402*002*3011*301*Q01*Q0100220022002*0021*901*D05*80100*01*K01000*2101*F010100*110111*D01*10*010101022*A01011100111005102*B01*0010111000112*B01*9012*B0*610*41*S02022*z0*V04*z0*z0*z0*G01*F0/b3?112&143&174&205/b4?201&202&171/b2?368&337&306&275/b1?214&183&152&121&90&59&28/b0?76/s0?45&44&75&74/s2?493&494&495&464&433/s1?43/f1?6/f3?518/f0?479/f7?113/f4?215/#replay=nmGTi8PB&1*4rlr0rdddrrr2urr*0u0r2l0ru*0l2u0lu*1lur2*1lullluuul1drr0rdllull2lu*5ruuu1ulldrrul0lllddrddrrdrdd*0rurrulldd1u0*0l).

I'll look into making a discord server if others are interested in these discussions. Otherwise direct messages are still handy

**Comment by JoeltheFox** [comment]
Another version of Ship Wreck for your consideration: [link](https://wolfesoftware.com/snakefall/#level=HyRr4JK1&19&36?*V04*K0111*S0*11*S0111*z0*z0*z0*z0*p0222*T0*02*H022*40*22*00*42*5022*30*32*G0222*10*52*701*20*02*10*62*501*W2/b0?542&578&614&579&615&617&581&618&582&546&547&616&220&257&294&293&292&256/s2?72&108&144&180&216&252&288&324&360&396&432&468&504&540&576/s3?545&544&543/)

**Comment by JoeltheFox** [comment]
[This](https://wolfesoftware.com/snakefall/#level=HyRr4JK1&29&32?*G02*202*R02*R02*201*D200222022*D01*402*002*3011*401*R01*R0100220022002*0021*A01*D05*90100*01*L01000*2101*G010100*110111*E01*10*010101022*B01011100111005102*C01*0010111000112*C01*9012*C0*610*41*T02022*z0*z0*z0*Z04*z0*N01*F0/f0?6/f1?117/b0?79/s0?47&46&78&77/s1?45/s2?509&510&511&479&447/f2?495/f3?535/f4?222/b1?221&189&157&125&93&61&29/b2?284&316&348&380/b3?116&148&180&212/b4?208&209&177/) version of Sawmill is solvable!

**Comment by jmdiamond3** [comment]
Solved! Awesome design. I'll work on shoring mine up and allowing only one solution, but you should definitely post this as an alternative version.
